src/main/scala/dpla/ingestion3/preMappingReports/XmlShredderReport.scala

```diff
@@ -43,7 +45,29 @@ class XmlShredderReport(val inputURI: String,
     val flattened = triples.withColumn("triples", explode(triples.col("triples")))
       .where("triples is not null")
 
-    flattened.select("triples.id", "triples.attribute", "triples.value")
+    // Each row in `flattenedAgain' has three columns: "id", "attribute", "value"
+    val flattenedAgain = flattened
+      .select("triples.id", "triples.attribute", "triples.value")
+
+    // Each row in the returned DataFrame has columns for "id" and for each
+    // attribute.  If there is more than one value for any attribute, it is
+    // concatenated with ",".
+    val grouped = flattenedAgain.groupBy("id")
+      .pivot("attribute")
+      .agg(concat_ws(",", collect_list(col("value"))))
```

Comment:
I'm not 100% sure we wanted to do this grouping + concatenation, but we'll see once we start examining the output.

Reply:
Sounds good, this is at least a starting place.

Reply:
Definitely.